source/merge-deep.d.ts

```diff
@@ -0,0 +1,182 @@
+import {ConditionalExcept} from './conditional-except';
+import {ArrayHead, ArrayTail} from './internal';
+
+/**
+MergeDeep options.
+
+@param strict Properties set to `undefined` value are skipped when set to `true` (default).
+
+@see MergeDeep
+*/
+export interface MergeDeepOptions {
+  strict?: boolean;
+}
+
+/**
+Represents an unknown record.
+*/
+type UnknownRecord = Record<string, unknown>;
+
+/**
+`Unwrap` is like `Simplify` it flatten the type output to improve type hints shown in editors.
+Unfortunately Simplify does not support the case where the type cannot be flattened.
```

Comment:
How does this handle types that cannot be flattened differently than `Simplify`?

Reply:
If you can't iterate in the type then it can't be flattened, a function for example. Here is the simplest case [TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyglgWzAGzgMxAHgCoD4oC8UA3gNoDSEI240cAdlANZUD2aU2AugFycVUakLgF8A3AFgAUNNCQoAVXoB3AE4BDMDlr4iQ6BAAewCPQAmAZygAlCAGNWqs5gvBVDAOYAaKAFd6TPSsyvS40gCQAPwkAiBQDMxsHPq8nLSxohF8+pIyUnLQAMLqyMgARup2TDi6UAAUAJSE+Ni5srRQAIKEsIgo6FjFpRVVmG6+ELhh+R0AQj1KapqYQ+WV1eOTuEA)

Reply:
Perfect, I will make a PR ;)